src/components/common/names-capture.vue

```diff
@@ -876,6 +858,55 @@ export default class NamesCapture extends NameRequestMixin {
       })
     }
   }
+
+  @Watch('isVisible')
+  private onVisibleChanged (val: boolean) {
+    if (val) {
+      // add event listener when this component is displayed
+      // eg, when user comes back from next tab
+      this.$el.addEventListener('keydown', this.handleKeydown)
+    } else {
+      // remove the event listener when this component is hidden
+      // eg, when user continues to next tab
+      this.$el.removeEventListener('keydown', this.handleKeydown)
+    }
+  }
+
+  @Watch('request_action_cd')
+  updateLocationOnAssumedName (newVal, oldVal) {
+    if (newVal === 'ASSUMED') {
+      if (this.location === 'BC') this.location = 'CA'
+    }
+  }
+
+  @Watch('entity_type_cd')
+  handleEntityType (newVal, oldVal) {
+    if (newVal === 'INFO') {
+      this.setPickEntityModalVisible(true)
+      this.entity_type_cd = oldVal
+    }
+  }
+
+  @Watch('location')
+  handleLocation (newVal, oldVal) {
+    if (newVal !== oldVal) {
+      this.setNRData({ key: 'xproJurisdiction', value: '' })
+    }
+    if (newVal === 'INFO') {
+      let type = this.entity_type_cd
+      this.setLocationInfoModalVisible(true)
+      this.$nextTick(function () {
```

Comment:
This is a strong code smell -- it's a workaround for something async not working right.

We should try to come back to this eventually and make it work right.